addons/l10n_id/models/res_bank.py

```diff
@@ -42,26 +46,47 @@ def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communi
 
         Getting content for the QR through calling QRIS API"""
         if qr_method == "id_qr":
+            invoice = self.env['account.move'].browse(self._context.get('qris_originating_invoice_id'))
+
+            # QRIS codes are valid for 30 minutes. To leave some margin, we will return the same QR code we already
+            # generated if the invoice is re-accessed before 25m. Otherwise, a new QR code is generated.
```

Comment:
I feel like this is done twice?

Reply:
Not exactly.
There is indeed two such checks:
- One when the cron is running, that discards the local data of the QR codes if they are already expired (> 30m, meaning that they can't be paid anymore)
- One when generating the QR to display in the portal invoice, which will return the already generated QR if it has been less than 25m since it was generated (in order to reduce the amount of QR statuses to check later on)